lib/commands/element.js

```diff
@@ -246,6 +252,75 @@ commands.clear = async function (el) {
   await retry(5, this.proxyCommand.bind(this), `/element/${el}/clear`, 'POST');
 };
 
+commands.getContentSize = async function (el) {
+  if (this.isWebContext()) {
+    throw new errors.NotYetImplementedError('Support for getContentSize for webcontext is not yet implemented. Please contact an Appium dev');
+  }
+
+  const type = await this.getAttribute('type', el);
+
+  if (type !== "XCUIElementTypeTable" &&
+      type !== "XCUIElementTypeCollectionView") {
+    throw new Error(`Can't get content size for type '${type}', only for ` +
+                    `tables and collection views`);
+  }
+  let locator = "*";
+  if (type === "XCUIElementTypeTable") {
+    // only find table cells, not just any children
+    locator = "XCUIElementTypeCell";
+  }
+
+  let contentHeight = 0;
+  let children = await this.findElOrEls(`class chain`, locator, true, el);
+  if (children.length === 1) {
+    // if we know there's only one element, we can optimize to make just one
+    // call to WDA
+    const rect = await this.getRect(_.first(children));
+    contentHeight = rect.height;
+  } else if (children.length) {
+    // otherwise if we have multiple elements, logic differs based on element
+    // type
+    switch (type) {
+      case "XCUIElementTypeTable": {
+        const firstRect = await this.getRect(_.first(children));
+        const lastRect = await this.getRect(_.last(children));
+        contentHeight = lastRect.y + lastRect.height - firstRect.y;
+        break;
+      }
+      case "XCUIElementTypeCollectionView": {
+        let elsInRow = 0;
+        let firstRect = await this.getRect(_.first(children));
+        let initialRects = [firstRect];
+        for (let i = 1; i < children.length; i++) {
+          const rect = await this.getRect(children[i]);
+          initialRects.push(rect);
+          if (rect.y !== firstRect.y) {
+            elsInRow = i;
+            break;
+          }
+        }
+        const spaceBetweenEls = initialRects[elsInRow].y - initialRects[elsInRow - 1].y - initialRects[elsInRow - 1].height;
+        const numRows = Math.ceil(children.length / elsInRow);
```

Comment:
can it be that `elsInRow` is still zero?

Reply:
aha, good catch. i'll initialize `elsInRow` to 1.